tests/ignite/handlers/test_terminate_on_nan.py

```diff
@@ -1,11 +1,28 @@
 import numpy as np
+import pytest
 import torch
 
 from ignite.engine import Engine, Events, State
 from ignite.handlers import TerminateOnNan
 
 
-def test_terminate_on_nan_and_inf():
+@pytest.mark.parametrize(
+    "state_output,should_terminate",
+    [
+        (1.0, False),
+        (torch.tensor(123.45), False),
+        (torch.asin(torch.randn(10,)), True),
+        (torch.asin(torch.randn(4, 4)), True),
+        ((10.0, 1.0 / torch.randint(0, 2, size=(4,)).type(torch.float), 1.0), True),
```

Comment:
Let's replace random data by deterministic:
```suggestion
        ((10.0, 1.0 / torch.tensor([1.0, 2.0, 0.0, 3.0]), 1.0), True),
```